crates/ark/src/lsp/diagnostics.rs

```diff
@@ -1331,31 +1343,70 @@ foo
             let document = Document::new(code, None);
 
             let diagnostics = generate_diagnostics(document.clone(), DEFAULT_STATE.clone());
-            assert_eq!(diagnostics.len(), 1);
-
-            let diagnostic = diagnostics.get(0).unwrap();
-            assert_eq!(diagnostic.range.start.line, 2);
-            insta::assert_snapshot!(diagnostic.message);
+            assert_eq!(diagnostics.len(), 0);
         })
     }
 
     #[test]
-    fn test_assignment_within_function_arguments() {
-        // TODO: `x` should be defined in the global env and there should not be a diagnostic here
+    fn test_assignment_within_call() {
+        // https://github.com/posit-dev/positron/issues/3048
+        // With our current approach we also incorrectly index symbols in calls
+        // with local scopes such as `local()` or `test_that()`. We prefer to be
+        // overly permissive than the opposite to avoid annoying false positive
+        // diagnostics.
         r_task(|| {
+            let code = "
+                list(x <- 1)
+                x
+            ";
+            let document = Document::new(code, None);
+
+            assert_eq!(
+                generate_diagnostics(document.clone(), DEFAULT_STATE.clone()).len(),
+                0
+            );
+
             let code = "
                 list({ x <- 1 })
                 x
             ";
+            let document = Document::new(code, None);
+
+            assert_eq!(
+                generate_diagnostics(document.clone(), DEFAULT_STATE.clone()).len(),
+                0
+            );
+        })
+    }
 
+    #[test]
+    fn test_no_symbol_diagnostics_in_calls() {
+        // For now we never check for missing symbols inside calls because we
+        // don't have a good way to deal with NSE in functions like `quote()` or
+        // `mutate()`.
+        r_task(|| {
+            let code = "
+                list(x)
+            ";
             let document = Document::new(code, None);
 
-            let diagnostics = generate_diagnostics(document.clone(), DEFAULT_STATE.clone());
-            assert_eq!(diagnostics.len(), 1);
+            assert_eq!(
+                generate_diagnostics(document.clone(), DEFAULT_STATE.clone()).len(),
+                0
+            );
 
-            let diagnostic = diagnostics.get(0).unwrap();
-            assert_eq!(diagnostic.range.start.line, 2);
-            insta::assert_snapshot!(diagnostic.message);
+            // Important to test nested case. We have a dynamic stack of state
+            // variable to keep track of whether we are in a call. The inner
+            // call should restore the outer state on exit.
+            let code = "
+                list(list(), x)
+            ";
```

Comment:
Maybe one more test of

```
list()
x
```

to make sure we get a diagnostic on `x`? that ensures we reset `in_call = false` on exit (similar to the existing test here, but a little different)